src/sif/motorcyclecost.cc

```diff
@@ -201,7 +201,19 @@ class MotorcycleCost : public DynamicCost {
    * @return  Returns true if access is allowed, false if not.
    */
   virtual bool Allowed(const baldr::NodeInfo* node) const {
-    return (node->access() & kMotorcycleAccess);
+    return (node->access() & (ignore_access_ ? kAllAccess : kMotorcycleAccess));
```

Comment:
branches are expensive and this method gets called A LOT. can we rewrite these to something like:

    return (node->access() & kMotorcycleAccess) || ignore_access_;

Reply:
these equations aren't equal. are we going to use the node in case `node->access()` is null ? right now it's not true

Reply:
yep i know they arent equal, but i made this comment addressing it: https://github.com/valhalla/valhalla/pull/2469/files#r482064088

im pretty sure we dont keep nodes in the graph that have `access==0` but maybe im wrong. maybe there are nodes which are impassable but the edges leading to t hem are passable... that is a good question but i think the spirit of "ignore_access" means i dont care whta access (even if none) exists, im going through it

Reply:
ok, got it